ast/compile_test.go

```diff
@@ -930,47 +930,111 @@ func TestCompilerExprExpansion(t *testing.T) {
 }
 
 func TestCompilerRewriteExprTerms(t *testing.T) {
-	module := `
-		package test
-
-		p { x = a + b * y }
-
-		q[[data.test.f(x)]] { x = 1 }
-
-		r = [data.test.f(x)] { x = 1 }
-
-		f(x) = data.test.g(x)
-
-		pi = 3 + .14
-
-		with_value { 1 with input as f(1) }
-	`
-
-	compiler := NewCompiler()
-	compiler.Modules = map[string]*Module{
-		"test": MustParseModule(module),
+	cases := []struct {
```

Comment:
> The weird ones are error cases 5 and 6. If the 2nd valid case is allowed, then arguably error cases 5 and 6 should be as well. I'm curious what your thoughts are here.

Been thinking this over. I tend to think the "right" behavior is what the 2nd case is doing. I would however be more inclined to opt for consistency, so whatever way we go we should make them all behave similarly.

I'll take a look into what we'd need to do to support cases 5 & 6

Reply:
After we chatted about this offline more the decision was to back up and just disallow calls to be in the rule args rather than rewriting them for now.